hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -883,6 +890,11 @@ public RecordListCheckPoint createRecordListCheckPoint() {
         return new RecordListCheckPoint(firstPreceding, lastFollowing);
     }
 
+    @Override
+    public void reclaimPreviouslyReservedThrottle(int n, HederaFunctionality function) {
+        networkUtilizationManager.leakUnusedThrottlePreviouslyReserved(n, function);
```

Comment:
We need this method to be a little more involved. 
   1. It should only leak capacity from the consensus throttle when the top-level transaction is a `CryptoTransfer`; because a top-level contract operation---`ContractCall`, `ContractCreate`, `EthereumTransaction`---will not have **reserved** capacity from the `CRYPTO_CREATE` throttle bucket. (I.e., if we always leak capacity here, then a `ContractCall` failing an auto-creation will effectively **reduce** the measured network utilization! Not what we want. 😁)
   2. We need to also leak capacity from the **frontend** throttle for **only the node that submitted this transaction**; c.f. mono-service [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/TransferLogic.java#L189). Reclaiming capacity from the frontend throttle is the main point of this PR, as otherwise client's will still get `BUSY` when submitting `CRYPTO_CREATE`s; even though we have detected that some preceding `CRYPTO_CREATE`'s actually failed, and reclaimed their network utilization from the consensus throttle.

Reply:
Isn't the implementation of `HandleContext.reclaimPreviouslyReservedThrottle()` okay being a bit generalized so that you can use it for different `HederaFunctionality` types and to add some restrictions on the current invocation of it in [TransferContextimpl](https://github.com/hashgraph/hedera-services/blob/10216-throttle-usage-auto-creation/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java#L102), based on your comments